### PR TITLE
[Mobile Payments] Handle exceptional cases for setting the receipt email

### PR DESCRIPTION
**Pull request by toupper**
Closes: #6570

Sorry for the larger PR, it got bloated after restoring the previous code and adding unit tests. I can split it into smaller PRs if that's better. The logic is however very straightforward.

### Description

As pointed [here](https://github.com/woocommerce/woocommerce-ios/issues/6570) , we removed the email receipt setting for all the cases in [this PR](https://github.com/woocommerce/woocommerce-ios/pull/6547). This however is not valid for some exceptional cases where we do have to set the email:

- WCPay plugin version is lower than 4.0.0
- WCPay is not active but Stripe is

If both WCPay and Stripe and installed and active, we do not send the receipt email.

With this PR we handle these cases.

### Changes
- Extract payments plugin information provider into its own class and use it in `CardPresentPaymentsOnboardingUseCase` and `PaymentCaptureOrchestrator` to avoid duplicated code
- Restore `Email` property wrapper, properties, and unit tests
- Implement new logic in `PaymentReceiptEmailParameterDeterminer`
- Use it in `PaymentCaptureOrchestrator`
- Unit tests

~~As you can see, I used `async/await` and throwable errors in `PaymentCaptureOrchestrator`. With it, we make the code simpler and more readable, as it could get very convoluted with nested completion callbacks. It is however only used and isolated in `PaymentCaptureOrchestrator` for this specific case, so it should not cause any undesired side effects in other scenarios.~~ As @jaclync pointed out, It is not possible until we update the Xcode version in our CI to Xcode 13.2

#### Prerequisites
- In order to test it, you have to have a Local Delivery (cash on delivery) order still processing. You can trigger that on your site before testing this flow in the app.
- Please make sure that you are able to test payments (more info [here](PdfdoF-D-p2)

#### Only WCPay installed, version lower than 4.0.0
Make sure that you have WooCommerce Payments plugin installed and active, but not Stripe.
1. Open App
2. Go to orders
3. Open processing order
4. Tap on Collect Payment
5. Proceed with payment with the reader and test card
6. Email information is set

#### WCPay installed, version 4.0.0 or higher
This version is not released yet (please correct me if I am wrong). The case is unit tested however in `PaymentReceiptEmailParameterDeterminerTests`: It should not set the email parameter

#### Only Stripe
Make sure that you have the Stripe plugin installed and active, but not WooCommerce Payments.
1. Open App
2. Go to orders
3. Open processing order
4. Tap on Collect Payment
5. Proceed with payment with the reader and test card
6. Email information is sent

#### Both plugins installed and active
Make sure that you have WooCommerce Payments and Stripe plugin installed and active, but not WooCommerce Payments.
1. Open App
2. Go to orders
3. Open processing order
4. Tap on Collect Payment
5. Proceed with payment with the reader and test card
6. Email information is not sent

### Screenshots

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Tools/PaymentsPluginsInfoProvider.swift
Comment (joshheald) [inline WooCommerce/Classes/Tools/PaymentsPluginsInfoProvider.swift]: I like that you've moved this logic out.

Did you consider putting it in the Yosemite layer? I don't think there's a problem with having it in the app layer (since that's where it came from) but it seems like it would fit better in Yosemite. That can be done later to avoid pushing the timescales out, if you have any concerns about that: this is a hot fix after all.

I'm still learning my way around the architecture so another opinion would be welcome! @ctarda do you have any thoughts on this?

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (joshheald) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: This logic looks correct to me.

I'd consider (maybe after the hotfix?) adding a function that returns an enum to `PaymentsPluginsInfoProvider`, something like
```
func activePlugin() -> PaymentPluginActivityState {
    // return .both if both are installed and active
    // if only one is active (even if both installed) return the active one
    // if both are installed, but none are active, return .wcPay
    // if one is installed, but not active return that.
    // if none are installed, return .none.
}

enum PaymentPluginActivityState {
    case .wcPay(active: Bool)
    case .stripe(active: Bool)
    case .both
    case .none
}
```

That way the stepping through the states can be shared as well.

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (joshheald) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: 👍 on clear naming

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (joshheald) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: I don't think we can use `async await` yet because we still support iOS 14... or am I out of the loop here?

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (joshheald) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: Can we rely on this already having happened during reader connection when taking payment? Seems like this might add an extra delay that isn't needed...

**Review by joshheald — COMMENTED** [review]
Thanks @toupper!

I've not tested this, just had a preliminary look over the code.

The logic all looks sound and I love that you've extracted the shared code. A few suggestions included here, but please question if you don't agree, you're closer to the work than me. Also, we can punt some or all of them to after the hotfix, since timing is important here.

Nice one 👍

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (kidinov) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: My 1ct - yes better to fatal, because this means that something really wrong goes in the code and the feature won't work anyway

Comment (kidinov) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: My 1ct - yes better to fatal, because this means that something really wrong goes in the code and the feature won't work anyway

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (kidinov) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: The logic looks good! jfyi I checked only this method

Comment (kidinov) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: The logic looks good! jfyi I checked only this method

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: It gained backward compatibility as you can see [here](https://www.swiftbysundell.com/special/swift-concurrency-backward-compatibility/), but depending on the decision we take [here](https://github.com/woocommerce/woocommerce-ios/pull/6585#discussion_r844910161) I can remove it because it wouldn't be (that) necessary

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: Yeah, that makes sense, I will update after I test it to double-check that it happened already

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (ctarda) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: A silent error is not ideal. I do not love the idea of crashing on purpose, but at the very least, we should log the reason why we stop execution, and we should try to inform the user of what is going on. 

In this particular case, we have seen this case happening in production. There have been users with both WCPay and the Stripe plugin installed, so we should guard against it properly.

File: WooCommerce/Classes/Tools/PaymentsPluginsInfoProvider.swift
Comment (ctarda) [reply WooCommerce/Classes/Tools/PaymentsPluginsInfoProvider.swift]: I am a bit confused about this struct. I do not see it in the project file in this diff, and I do not see it in the Xcode project tree if I check out the branch. Is this actually being used?

Comment (joshheald) [reply WooCommerce/Classes/Tools/PaymentsPluginsInfoProvider.swift]: It got renamed, sorry, I was going commit by commit when I reviewed. From the next commit it's called `PaymentsPluginsDataProvider`

Comment (toupper) [reply WooCommerce/Classes/Tools/PaymentsPluginsInfoProvider.swift]: Exactly, as @joshheald mentioned, I renamed to `PaymentsPluginsDataProvider`, a more meaningful name IMHO

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (kidinov) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: > In this particular case, we have seen this case happening in production. There have been users with both WCPay and the Stripe plugin installed, so we should guard against it properly.

Please correct me if on iOS it's not the case. We should not build `PaymentIntent` when IPP is not available for the site, but instead, we should show the proper error screen that I think is already in place for iOS too and user won't be able to start collecting payment if 2 plugins are active?

If this is correct, then returning null from here will end up in the error/nonfuncinalal payment somewhere later and we won't probably see a clear picture of why it happens

**Fatal approach**
* Quick feedback loop from sentry
* Crash stack clear says what and where in the code it happened 

**Logging and returning null**
* People don't check logs mostly so the issue will be in place longer
* Even if users will start complaining that IPP doesn't work in some cases it will be not easy to figure out why

The result is the same - IPP doesn't work in both approaches, but with the first approach we will fix it faster, spending less time on investigation

Comment (ctarda) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: We still have the issue with Order Details: pdfdoF-yu-p2

It is true that we should not start creating a PaymentIntent if the store and the order is not eligible, but the fact is that, due to an oversight on our side, it is possible. 

Crashing when the merchant is trying to capture a payment does not sound like the best way to communicate that something is not quite right. Showing an error message and aborting the attempt to capture a payment sounds to me like a better way to fail.

And if we log the error we can still look at the application logs to trace exactly what happened.

Comment (joshheald) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: 😅 I knew this would be a spicy one. It was my suggestion on our sync call yesterday that @toupper consider crashing here. 

I'm not stirring up trouble deliberately, I'm really not sure which way we should go for all the above reasons.

Probably in this case, crashing because they have both installed is overzealous – all we're using the information for is to set an optional email parameter. So returning `nil` is not so bad, the things which people care about most (the payment working and being correct) will happen even if we set the email to nil incorrectly.

If we want to crash for having both plugins installed after the connection to the reader has happened, perhaps that would be better done (in other work) at the point where we decide which plugin to use...

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (jaclync) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: this looks nice! my understanding of using async/await is that we have to wait for CI to upgrade the Xcode version to at least 13.2. Right now it's on 13.0:

https://github.com/woocommerce/woocommerce-ios/blob/6242f86223d2329d45a8b6702e6b443d9622005b/.circleci/config.yml#L18

Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: Thanks, @jaclync for pointing that out! In that case, we cannot use it, because Xcode 13.0 only allows `async/await` for iOS 15. I removed it, a pity 😅

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: After some checking I think it is better to leave this synching. When paying directly in the orders tab without going through the settings onboarding we do not synch the plugins. We do when we start the app, but It could be that the plugins state changed during the app execution.

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: Thank you all for the feedback 🎉 ! As it seems we have these options:

1. Fatal error and crash: Pros: Quick and meaningful feedback. Cons: Not a good user experience especially in a critical user case (payment).
2. Payment Error:
           - Silently: Pros: Less code and complexity to implement Cons: Not a good experience for the user, who might not know what is happening.
           - With a UI Message: Pros: Good user experience, communicate to the user what is happening. Cons: Payment fails (is it necessary?)
3. Go with the payment but return nil receipt email: Pros: Less code and complexity, the payment goes on. Cons: Would the payment work if both of them are installed and active? If that is the case, I would go for this one for this hotfix and create an issue to properly handle this case on the payment in the future.

Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: Moved discussion here pdfdoF-CS-p2

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: Good idea! I created an issue [here](https://github.com/woocommerce/woocommerce-ios/issues/6610) for this

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift]: The outcome of the discussion:
- For the receipt email case, we do not send it if we have WCPay and Stripe both installed and active at the same time. We do not fail or crash.
- We implement the [onboarding checks](https://href.li/?https://github.com/woocommerce/woocommerce-ios/issues/6607) in another PR

File: WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift
Comment (jaclync) [inline WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift]: nit: could these be private?

File: WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift
Comment (jaclync) [inline WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift]: nit: could this also be private since it's not in the protocol?

File: WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift
Comment (jaclync) [inline WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift]: nit: since we are most likely going to use this for in-person refunds for 🇨🇦 , do you think a more generic name like `CardPresentPluginsDataProvider` would be good?

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift]: super nit:
```suggestion
    ///   - onCompletion: closure invoked with the result of the inquiry, containing the email (if any) or error
```

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift]: I was wondering about this case - if both plugins are active, how is the receipt email sent when the WCPay version is below the minimum version (4.0.0)?

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift:223]: nice!

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift:24]: 💭 I was thinking about whether we want to sync the plugins again here - it does ensure that the plugins are up-to-date, but if it fails it might result in an error state for the whole payment flow. It might also add some time from the API request, for each transaction. Since the plugins usually stay the same, I wonder if we could make the syncing less often at least 🤔 

Right now, the plugins are synced on launch and store switch:

https://github.com/woocommerce/woocommerce-ios/blob/db2f2934697a3a852d3b7bd07be7476790d6d82c/WooCommerce/Classes/Yosemite/DefaultStoresManager.swift#L456

In any case, we can always change the syncing later.

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:23]: nit: maybe also call super here?
```suggestion
        stores = nil
        super.tearDown()
```

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:29]: nit: we usually add `Given/When/Then` comments to test cases, so that it's easier to read the code for each part 🙂 (similarly for other test cases in this file)

```suggestion
        // Given
        let order = Order.fake()
```

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:37]: nit: from the comment above
```suggestion
        // When
        let result: Result<String?, Error> = waitFor { promise in
```

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:44]: nit: from the comment above
```suggestion
        // Then
        guard case let .success(email) = result else {
```

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift]: super nit:
```suggestion
    func test_when_only_WCPay_is_active_and_version_is_equal_to_minimum_that_sends_email_then_returns_nil() {
```

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift]: nit: do we also want to pass a valid email to the order like in other test cases?

**Review by jaclync — APPROVED** [review]
Nice work on creating new structs for the plugin data and determining the email with test cases! Apology for the late review, it took me some time to test the scenarios ⚙️ The changes LGTM, I only had nit suggestions and a non-blocking question 😄 

For testing:
- Only WCPay installed, version lower than 4.0.0 ✅
- WCPay installed, version 4.0.0 or higher ❓ I wasn't able to test this, I tried downloading the zip from the [WCPay repo](https://github.com/Automattic/woocommerce-payments) but wasn't able to activate this on my site with the following error:
```
Warning: require_once(/srv/htdocs/wp-content/plugins/woocommerce-payments-develop/vendor/autoload_packages.php): failed to open stream: No such file or directory in /srv/htdocs/wp-content/plugins/woocommerce-payments-develop/woocommerce-payments.php on line 27

Fatal error: require_once(): Failed opening required '/srv/htdocs/wp-content/plugins/woocommerce-payments-develop/vendor/autoload_packages.php' (include_path='/:.') in /srv/htdocs/wp-content/plugins/woocommerce-payments-develop/woocommerce-payments.php on line 27
```
- Only Stripe ✅
- Both plugins installed and active: the Collect Payment CTA isn't shown in this case pdfdoF-CS-p2#comment-1248

File: WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift
Comment (toupper) [reply WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift]: Yeah, thanks for the suggestion, I renamed it

File: WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift
Comment (toupper) [reply WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift]: Yup, done :)

File: WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift
Comment (toupper) [reply WooCommerce/Classes/Tools/PaymentsPluginsDataProvider.swift]: Indeed, good catch, done :)

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift]: Nice! yeah, fixed

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:23]: 👍

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:29]: Good idea! added

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:37]: 👌

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift:44]: 🆗 😃

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift]: NIce one! corrected

File: WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminerTests.swift]: Yup, I thought that `fake` initializes the objects with a value, but for optional it returns nil. Good point!

File: WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift
Comment (toupper) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift:24]: Thanks for the pointing that out @jaclync :) We also thought about it [here](https://github.com/woocommerce/woocommerce-ios/pull/6585#discussion_r846008429)

>  it does ensure that the plugins are up-to-date, but if it fails it might result in an error state for the whole payment flow.

Actually, if there is an error syncing we do not stop the whole payment flow, we just returned nil for email in `PaymentCaptureOrchestrator`:

```
paymentReceiptEmailParameterDeterminer.receiptEmail(from: order) { result in
            var receiptEmail: String?
            if case let .success(email) = result {
                receiptEmail = email
            }
```
About syncing, it is true that it creates an extra request, but I wanted the class to be totally independent and avoid any assumptions about previous syncing (e.g if someone removes the syncing at launch this would break without us knowing it)
Furthermore, users can change the plugins configuration while the app is running, an unlikely case but possible anyways. I tried the request with bad network conditions, and still, it doesn't take very long.
As you said, we can always remove it if we detect that this is suboptimal, e.g when adding the onboarding checks to the payment collection in orders. I will add a note in that issue 🚀

Comment (jaclync) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift:24]: Sounds great, thanks for the detailed explanation!

**Review by jaclync — APPROVED** [review]
Thanks for the updates @toupper, still LGTM and please feel free to merge for release 8.9! 🚀

**Pull request merged**
